projects/telemetry/src/nmea.c

```diff
@@ -0,0 +1,115 @@
+#include "nmea.h"
+#include <string.h>
+#include <stdint.h>
+#include "log.h"
+NMEAResult parse_nmea_sentence(const uint8_t *rx_arr, size_t len) {
+  LOG_DEBUG("NMEA Parser recieved data with len %zu\n", len);
+  NMEAResult r = {0};
+
+  // m_id will keep track of which sentence type we are currently operating on
+  NMEAMessageID m_id = 0;
+  for (uint8_t i = 0; i < len; i++) {
+    LOG_DEBUG("%c", rx_arr[i]);
+  }
+
+  // Should use context somewhere here to make sure the message is for us
+
+  if (len < 1 && rx_arr[0] != '$') {
+    LOG_WARN("First character of a NMEA sentence should be a $, it is a %c", rx_arr[0]);
+  }
+  if (len < 3 && (rx_arr[1] != 'G' || rx_arr[2] != 'P')) {
+    LOG_WARN("NMEA sentence should begin with GP. They are %c%c", rx_arr[1], rx_arr[2]);
+  }
+  // Array index 3 should be 0
+  uint8_t message_id[4] = {0};
+  for (uint32_t i = 3; i < 6; i++) {
+    if (rx_arr[i] == ',') {
+     break;
```

Comment:
Weird spacing? Run `make format` before pushing.

Reply:
Did this